src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -90,21 +89,18 @@ export async function makeCsharpConfig(config: SamLaunchRequestArgs): Promise<Sa
     }
 
     const baseBuildDir = await makeBuildDir()
-    const template: CloudFormation.Template = await CloudFormation.load(config.samTemplatePath)
-    const resource = await CloudFormation.getResourceFromTemplateResources({
-        templateResources: template.Resources,
-        handlerName: config.handlerName,
-    })
-    const codeUri = getCodeUri(resource, vscode.Uri.parse(config.samTemplatePath!!))
+    const template = getTemplate(config)
+    const resource = getTemplateResource(config)
+    const codeUri = getCodeRoot(config.workspaceFolder, config)
```

Comment:
cannot assume that `config.samTemplatePath` is defined here; instead use `getCodeRoot()` to get the info for both target=code and target=template cases.